pkg/controllers/nodetemplate/suite_test.go

```diff
@@ -0,0 +1,129 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter-core/pkg/operator/injection"
+	"github.com/aws/karpenter-core/pkg/operator/options"
+	"github.com/aws/karpenter-core/pkg/operator/scheme"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/aws/karpenter-core/pkg/apis/config/settings"
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+	coretest "github.com/aws/karpenter-core/pkg/test"
+	. "github.com/aws/karpenter-core/pkg/test/expectations"
+	"github.com/aws/karpenter/pkg/apis"
+	coresettings "github.com/aws/karpenter/pkg/apis/config/settings"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/controllers/nodetemplate"
+	"github.com/aws/karpenter/pkg/fake"
+	"github.com/aws/karpenter/pkg/provider"
+	"github.com/aws/karpenter/pkg/test"
+)
+
+var ctx context.Context
+var env *coretest.Environment
+var fakeEC2API *fake.EC2API
+var opts options.Options
+var subnetProvider *provider.SubnetProvider
+var securityGroupProvider *provider.SecurityGroupProvider
+var nodeTemplate *v1alpha1.AWSNodeTemplate
+var controller corecontroller.Controller
+var settingsStore coretest.SettingsStore
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "AWSNodeTemplateStatusController")
+}
+
+var _ = BeforeSuite(func() {
+	env = coretest.NewEnvironment(scheme.Scheme, apis.CRDs...)
+	settingsStore = coretest.SettingsStore{
+		coresettings.ContextKey: coretest.Settings(),
+		coresettings.ContextKey: test.Settings(),
+	}
+	ctx = settingsStore.InjectSettings(ctx)
+
+	fakeEC2API = &fake.EC2API{}
+	subnetProvider = provider.NewSubnetProvider(fakeEC2API)
+	securityGroupProvider = provider.NewSecurityGroupProvider(fakeEC2API)
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = BeforeEach(func() {
+	ctx = injection.WithOptions(ctx, opts)
+	settingsStore = coretest.SettingsStore{
+		settings.ContextKey:     coretest.Settings(),
+		coresettings.ContextKey: test.Settings(),
+	}
+	ctx = settingsStore.InjectSettings(ctx)
+
+	nodeTemplate = &v1alpha1.AWSNodeTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: coretest.RandomName(),
+		},
+		Spec: v1alpha1.AWSNodeTemplateSpec{
+			AWS: v1alpha1.AWS{
+				SubnetSelector:        map[string]string{"*": "*"},
+				SecurityGroupSelector: map[string]string{"*": "*"},
+			},
+		},
+	}
+	nodeTemplate.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   v1alpha1.SchemeGroupVersion.Group,
+		Version: v1alpha1.SchemeGroupVersion.Version,
+		Kind:    "AWSNodeTemplate",
+	})
+	controller = nodetemplate.NewController(env.Client, fakeEC2API, subnetProvider, securityGroupProvider)
+
+	fakeEC2API.Reset()
+})
+
+var _ = AfterEach(func() {
+	ExpectCleanedUp(ctx, env.Client)
+})
+
+var _ = Describe("AWSNodeTemplateStatusController", func() {
```

Comment:
Should we include a unit test that ensures this is updated if the AWSNodeTemplate changes?